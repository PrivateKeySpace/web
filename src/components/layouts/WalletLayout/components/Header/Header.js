import React from 'react'
import { compose, setDisplayName, pure } from 'recompose'
import { css } from 'aphrodite/no-important'
import tx from '../../../../../theme/styles'
import sx from './styles'
import { Menu } from './components'

const displayName = 'components/layouts/WalletLayout/components/Header'

const enhance = compose(
  setDisplayName(displayName),
  pure
)

function Header (props) {
  const { content } = props
  return (
    <div className={css(tx.p5, tx.pt3, tx.flex, tx.flexDirectionColumn, tx.width100, sx.container)}>
      <div className={css(tx.flex, tx.flexDirectionRow, tx.alignItemsCenter, tx.width100, sx.navbar)}>
        <h2 className={css(tx.textTransformUppercase)}>
          PKS Wallet
        </h2>
        <div className={css(tx.mr4, tx.fontSize1, sx.notifications)}>
          <span className='lnr lnr-alarm' />
          <div className={css(sx.notificationCounterWrap)}>
            <div className={css(tx.flex, tx.alignItemsCenter, tx.justifyContentCenter, sx.notificationCounter)}>5</div>
          </div>
        </div>
        <div className={css(sx.separator)} />
        <div className={css(tx.ml4, tx.flex, tx.flexDirectionColumn, tx.alignItemsFlexEnd)}>
          <div className={css(tx.textTransformUppercase, sx.deviceName)}>
            Trezor Label
          </div>
          <div className={css(tx.textTransformUppercase, sx.deviceStatus)}>
            connected
          </div>
        </div>
        <Menu />
      </div>
      <div className={css(tx.pb3, sx.content)}>
        {content}
      </div>
    </div>
  )
}

export default enhance(Header)
