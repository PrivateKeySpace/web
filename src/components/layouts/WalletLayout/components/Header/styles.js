import { StyleSheet } from 'aphrodite/no-important'
import { colors } from '../../../../../theme/constants'

const styles = StyleSheet.create({
  container: {
    height: 175,
    backgroundColor: colors.darkBg,
    color: colors.white
  },
  navbar: {
    height: 36
  },
  pullRight: {
    marginLeft: 'auto'
  },
  notifications: {
    position: 'relative'
  },
  notificationCounterWrap: {
    position: 'absolute',
    bottom: -5,
    right: -10,
    width: 20,
    height: 20,
    borderRadius: 20,
    backgroundColor: 'rgba(255, 255, 255, 0.2)'
  },
  notificationCounter: {
    backgroundColor: colors.danger,
    width: 16,
    height: 16,
    borderRadius: 16,
    margin: 2,
    fontSize: 10,
    lineHeight: '10px'
  },
  separator: {
    width: 1,
    height: '75%',
    backgroundColor: colors.gray
  },
  deviceName: {
    color: colors.white,
    fontSize: 14
  },
  deviceStatus: {
    color: colors.white,
    opacity: 0.5,
    fontSize: 10
  },
  dropdownButton: {
    color: colors.white,
    opacity: 0.5
  },
  content: {
    marginTop: 'auto'
  }
})

export default styles
