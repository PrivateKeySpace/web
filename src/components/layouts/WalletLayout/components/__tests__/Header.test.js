import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import Header from '../Header'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders without crashing and matches snapshot', () => {
  const wrapper = render(
    <Header />
  )
  expect(wrapper).toMatchSnapshot()
})
