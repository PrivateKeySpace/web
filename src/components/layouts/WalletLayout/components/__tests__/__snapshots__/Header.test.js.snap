// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<div
  class="p4_1x929gn-o_O-pt3_1gv8f73-o_O-flex_jro6t0-o_O-flexDirectionColumn_1jtvojh-o_O-width100_zdxht7-o_O-container_1hoplq"
>
  <div
    class="flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-width100_zdxht7-o_O-navbar_1u75k1v"
  >
    <div
      class="textTransformUppercase_4efw5a-o_O-fontSize1_3kvou9"
    >
      PKS Wallet
    </div>
    <div
      class="mr4_1wkefyg-o_O-fontSize1_3kvou9-o_O-notifications_ls0e43"
    >
      <span
        class="lnr lnr-alarm"
      >
        
      </span>
    </div>
    <div
      class="separator_xrxmxy"
    >
      
    </div>
    <div
      class="ml4_2ffngz-o_O-flex_jro6t0-o_O-flexDirectionColumn_1jtvojh-o_O-alignItemsFlexEnd_1f2mvon"
    >
      <div
        class="textTransformUppercase_4efw5a-o_O-deviceName_1pe4kle"
      >
        Trezor Label
      </div>
      <div
        class="textTransformUppercase_4efw5a-o_O-deviceStatus_z7nj5y"
      >
        connected
      </div>
    </div>
    <div
      class="flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-px2_6ebohg-o_O-ml1_1dn7v29-o_O-height100_1yfus1e-o_O-dropdownButton_1gb1rbb"
    >
      <span
        class="lnr lnr-chevron-down"
      >
        
      </span>
    </div>
  </div>
  <div
    class="pb3_1a70m23-o_O-content_14ddvtv"
  >
    
  </div>
</div>
`;
